op-batcher/metrics/metrics.go

```diff
@@ -393,6 +394,13 @@ func (m *Metrics) RecordL2BlockInPendingQueue(block *types.Block) {
 	atomic.AddInt64(&m.pendingDABytes, int64(daSize))
 }
 
+func (m *Metrics) RecordPendingBlockPruned(block *types.Block) {
+	daSize, rawSize := estimateBatchSize(block)
+	m.pendingBlocksBytesTotal.Add(-1.0 * float64(rawSize))
+	m.pendingBlocksBytesCurrent.Add(-1.0 * float64(rawSize))
+	atomic.AddInt64(&m.pendingDABytes, -1*int64(daSize))
+}
```

Comment:
So this is like `RecordL2BlockInChannel` but also decreasing the total, right? And we want to subtract from the total because it was a pruned block from a previous batcher instance?

It would be helpful to add a short comment above the metric (or at the interface) since it's an unintuitive edge case.